app/models/katello/sync_plan.rb

```diff
@@ -23,13 +23,18 @@ class SyncPlan < Katello::Model
     validates :interval, :inclusion => {:in => TYPES}, :allow_blank => false
     validates :enabled, :inclusion => [true, false]
     validate :validate_sync_date
+    validate :product_enabled
     validates_with Validators::KatelloNameFormatValidator, :attributes => :name
 
     scoped_search :on => :name, :complete_value => true
     scoped_search :on => :organization_id, :complete_value => true, :only_explicit => true, :validator => ScopedSearch::Validators::INTEGER
     scoped_search :on => :interval, :complete_value => true
     scoped_search :on => :enabled, :complete_value => true
 
+    def product_enabled
+      errors.add :base, _("Can not add disabled products to sync plan") unless self.products.all?(&:enabled?)
```

Comment:
Ah, yes. And these errors should not be added to :base, but :product. Something like this:

```rb
products.each do |product|
  errors.add :product, _("#{product.name} can not be disabled.")
end
```